micmac_li3ds/import_autocal.py

```diff
@@ -56,177 +46,186 @@ def get_parser(self, prog_name):
+            self.transfos.append(orintglob)
+            target = source
+
+        distos = reversed(xmlutil.children(node, 'CalibDistortion'))
+        for i, disto in enumerate(distos):
+            source = Referential.undistorted(self.sensor, referential, i)
+            distortion = Transfo.distortion(source, target, transfo, disto, i)
+            self.referentials.append(source)
+            self.transfos.append(distortion)
+            target = source
+
+        source = Referential.camera(self.sensor, referential)
+        pinhole = Transfo.pinhole(source, target, transfo, node)
+        self.referentials.append(source)
+        self.transfos.append(pinhole)
+        self.camera = source
+
+        self.transfotree = Api.Transfotree(self.transfos, transfotree)
+        super().__init__()
+
+
+class Sensor(Api.Sensor):
+    def __init__(self, sensor, node):
+        specs = {'image_size': xmlutil.child_floats_split(node, 'SzIm')}
+        super().__init__(sensor, type='camera', specifications=specs)
+
+
+class Referential:
+    def distorted(sensor, referential):
+        description = 'origin: top left corner of top left pixel, ' \
+                      '+XY: raster pixel coordinates, ' \
+                      '+Z: inverse depth (measured along the optical axis). ' \
+                      '{description}'
+        return Api.Referential(
+            sensor, referential,
+            name='distorted',
+            description=description.format(**referential),
+            root=True,
+        )
+
+    def raw(sensor, referential):
+        description = 'origin: top left corner of top left pixel, ' \
+                      '+XY: raster pixel coordinates, ' \
+                      '+Z: inverse depth (measured along the optical axis). ' \
+                      '{description}'
+        return Api.Referential(
+            sensor, referential,
+            name='raw',
+            description=description.format(**referential),
+        )
+
+    def undistorted(sensor, referential, i):
+        description = 'origin: top left corner of top left pixel, ' \
+                      '+XY: raster pixel coordinates, ' \
+                      '+Z: inverse depth (measured along the optical axis). ' \
+                      '{description}'
+        return Api.Referential(
+            sensor, referential,
+            name='undistorted[{}]'.format(i),
+            description=description.format(**referential),
+        )
+
+    def camera(sensor, referential):
+        description = 'origin: camera position, ' \
+                      '+X: right of the camera, ' \
+                      '+Y: bottom of the camera, ' \
+                      '+Z: optical axis (in front of the camera), ' \
+                      '{description}'
+        return Api.Referential(
+            sensor, referential,
+            name='camera',
+            description=description.format(**referential),
+        )
+
+
+class Transfo:
```

Comment:
You don't need a class here. Why not just defining `create_pinhole`, `create_orintglob_transfo` and `create_distortion_transfo` functions instead?